File: class12/cpp/AP1210-PrecedenciaOperadores.cpp
#include <iostream>
int main() {
    int a = 10, b = 5, c = 2;
    std::cout << "a=" << a << ", b=" << b << ", c=" << c << std::endl;
    std::cout << "a + b * c = " << a+b*c << "   (multiplicação antes da soma)" << std::endl;
    std::cout << "(a + b) * c = " << (a+b)*c << " (parênteses alteram a ordem)" << std::endl;
    std::cout << std::boolalpha;
    std::cout << "a > b && b > c = " << (a>b && b>c) << " (relacionais antes do lógico)" << std::endl;
    std::cout << "! (a < b) || c > 0 = " << (!(a<b) || c>0) << " (NOT antes do OR)" << std::endl;
    return 0;
}

File: class12/python/AP1210-PrecedenciaOperadores.py
# PrecedenciaOperadores — Aula 12
# Demonstra a precedência de operadores aritméticos, relacionais e lógicos

a, b, c = 10, 5, 2
print(f"a={a}, b={b}, c={c}")
print(f"a + b * c = {a + b * c}   (multiplicação antes da soma)")
print(f"(a + b) * c = {(a + b) * c} (parênteses alteram a ordem)")
print(f"a > b and b > c = {a > b and b > c} (relacionais antes do lógico)")
print(f"not a < b or c > 0 = {not a < b or c > 0} (NOT antes do OR)")
